lib/pbio/src/light/animation.c

```diff
@@ -117,10 +117,15 @@ PROCESS_THREAD(pbio_light_animation_process, ev, data) {
         PROCESS_WAIT_EVENT_UNTIL(ev == PROCESS_EVENT_TIMER);
         struct etimer *timer = data;
         pbio_light_animation_t *animation = PBIO_CONTAINER_OF(timer, pbio_light_animation_t, timer);
-        clock_time_t interval = animation->next(animation);
-        if (pbio_light_animation_is_started(animation)) {
-            etimer_reset_with_new_interval(&animation->timer, interval);
+
+        // Like all etimer events, only handle expired etimers. Need this check
+        // in case the timer was stopped or reset before this event was handled.
+        if (!etimer_expired(&animation->timer) || !pbio_light_animation_is_started(animation)) {
```

Comment:
```suggestion
        if (!etimer_expired(timer) || !pbio_light_animation_is_started(animation)) {
```

We already have a pointer to the timer.

Reply:
Thank you. I suppose then we can use the conventional pattern as we do everywhere else:

```c
PROCESS_WAIT_EVENT_UNTIL(ev == PROCESS_EVENT_TIMER && etimer_expired(data));
```